### ISSUE TITLE
Add search definition and declaration for symbols from stdlib

### ISSUE THREAD

**Issue by nikpivkin** [pherrymason/c3-lsp#47]
Currently, searching for definitions for symbols from `stdlib` is not supported (`TextDocumentDefinition` and `TextDocumentDeclaration` handlers). I don't mind helping to implement this functionality. I see the following steps for implementation:
- When indexing `stdlib`, add ranges and document uri to all definitions
- Determine path to `stdlib` at LSP server startup
- Allow users to override the path to `stdlib`
- If the symbol is in `stdlib`, then load its module and resolve the absolute file path.

Let me know if I missed something.

**Comment by pherrymason** [comment]
That is a good solution, but it will require the user to have the spurces of the stdlib.

There is another option and is to include the stdlib with the lsp server and use the functionality 'virtual document' of the lsp proto ol. Haven't look at the specification deeply, but as I understood the lsp can deliver a reference to a document+ the document itself and the editor should display it.

**Comment by nikpivkin** [comment]
A future 3.18 version of LSP will add a [workspace/textDocumentContent](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/lsp/3.18/workspace/textDocumentContent.md) method that provides document content. Another option is to add a custom handler to return virtual documents, but the `github.com/tliron/glsp` package doesn't allow registering custom handlers (but [there](https://github.com/tliron/glsp/pull/32) is a PR for adding this).

**Comment by pherrymason** [comment]
In that case, I would analyze what's cheaper to implement:

- A new option for the LSP to specify path to the stdlib sources.
- Implement either the custom handler or [workspace/textDocumentContent](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/lsp/3.18/workspace/textDocumentContent.md) on top of `tliron/glsp`

**Comment by pherrymason** [comment]
Implemented in branch `file-config`.

Create a `c3lsp.json` file in your c3 project:

```
{
    "C3": {
        "version": "0.6.1",
        "stdlib-path": "<YOUR PATH TO STDLIB>/c3c/lib/std"
    },
    "Diagnostics": {
        "enabled": true,
        "delay": 2000
    }
}
```

**Comment by pherrymason** [comment]
Implemented in v0.3.0
